orchestrate.py

```diff
@@ -244,8 +256,10 @@ def main():
     handlers = [
         (r"/", LoadingHandler),
         (r"/spawn/?(/user/\w+(?:/.*)?)?", SpawnHandler),
+        (r"/spawn/((?:notebooks|tree)(?:/.*)?)", SpawnHandler),
```

Comment:
Instead of using notebooks and tree in the path, what if it's just a `redirect=` query arg on the URL that SpawnHandler and LoadingHandler respects? Then it's not notebook specific and satisfies @rgbkrk's comment in the PR.

The only thing to be careful about in that case is that the redirect should be parsed down to a path and only a path. Allowing anything in there and redirect the browser to it = XSS for sure.